rmw_connextdds_common/src/common/rmw_context.cpp

```diff
@@ -254,16 +254,36 @@ rmw_connextdds_initialize_discovery_options(
     }
     /* Support for ROS_AUTOMATIC_DISCOVERY_RANGE == LOCALHOST
       -----------------------------------------
-      Make sure that the participant is not listening on any multicast address. */
+    */
     if (RMW_AUTOMATIC_DISCOVERY_RANGE_LOCALHOST ==
       ctx->discovery_options->automatic_discovery_range)
     {
+      // Make sure that the participant is not listening on any multicast address.
       if (!DDS_StringSeq_ensure_length(
           &dp_qos.discovery.multicast_receive_addresses, 0, 0))
       {
         RMW_CONNEXT_LOG_ERROR_SET("failed to resize multicast_receive_addresses")
         return RMW_RET_ERROR;
       }
+      // Increase maximum participant ID
+      // This controls the number of participants that can be discovered on a single host,
+      // which is roughly equivalent to the number of ROS 2 processes.
+      // If it's too small then we won't connect to all participants.
+      // If it's too large then we will send a lot of announcement traffic.
+      // The default number here is picked arbitrarily.
+      const rmw_peer_address_t localhost_peers[2] = {
+        "32@builtin.udpv4://127.0.0.1",
+        "32@builtin.shmem://",
+      };
+      const auto rc2 = rmw_connextdds_extend_initial_peer_list(
```

Comment:
These peers should be added to the list only if it has not been already customized by the user in the environment variable and/or XML Qos, otherwise they might shadow custom values they configured

Reply:
Thats a good point. The current strategy for enabling users to control these settings is to set the range to `RMW_AUTOMATIC_DISCOVERY_RANGE_SYSTEM_DEFAULT`. It looks like maybe line 242 of this PR should be `else if (ctx->discovery_options->automatic_discovery_range != ..._SYSTEM_DEFAULT)` so that we don't try to extend the peer list in that case.

Reply:
Skipped modifying initial peers list when automatic discovery range is SYSTEM_DEFAULT in c501a76